modules/apps/frontend-taglib/frontend-taglib-clay-sample-web/src/main/java/com/liferay/frontend/taglib/clay/sample/web/internal/filter/CharlieDateRangeClayTableDataSetFilter.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.sample.web.internal.filter;
+
+import com.liferay.frontend.taglib.clay.data.set.filter.BaseDateRangeClayDataSetFilter;
+import com.liferay.frontend.taglib.clay.data.set.filter.ClayDataSetFilter;
+import com.liferay.frontend.taglib.clay.data.set.filter.DateClayDataSetFilterItem;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.Calendar;
+
+/**
+ * @author Javier de Arcos
+ */
+@Component(
+	property = "clay.data.set.display.name=dataSetDisplay",
+	service = ClayDataSetFilter.class
+)
+public class CharlieDateRangeClayTableDataSetFilter
+	extends BaseDateRangeClayDataSetFilter {
+
+	@Override
+	public String getId() {
+		return "charlie";
+	}
+
+	@Override
+	public String getLabel() {
+		return "Charlie Date Range";
+	}
+
+	public DateClayDataSetFilterItem getMaxDateClayDataSetFilterItem() {
+		Calendar calendar = Calendar.getInstance();
+
+		return new DateClayDataSetFilterItem(
+			calendar.get(Calendar.DAY_OF_MONTH),
+			calendar.get(Calendar.MONTH) + 1, calendar.get(Calendar.YEAR));
+	}
```

Comment:
This is fine for this PR, but a general note: this looks like over-engineering. Why not
```java
public Date getMaxDate() {
	return ...;
}
```

Reply:
We can annotate this point to try to refine it

Reply:
Yep. But this is definitely for a separate task. For now, let's just use the structure as it it.